core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/sync/TargetImageAwareMixedPackageVersionService.java

```diff
@@ -97,6 +105,8 @@ private String getCmVersion(Map<String, String> packageVersions) {
     }
 
     private void sendNotification(Long stackId, ResourceEvent resourceEvent, List<String> args) {
-        eventService.fireCloudbreakEvent(stackId, UPDATE_IN_PROGRESS.name(), resourceEvent, args);
+        Optional<StackStatus> currentStatus = stackStatusService.findFirstByStackIdOrderByCreatedDesc(stackId);
+        Status status = currentStatus.isPresent() && UPDATE_FAILED.equals(currentStatus.get().getStatus()) ? UPDATE_FAILED : UPDATE_IN_PROGRESS;
+        eventService.fireCloudbreakEvent(stackId, status.name(), resourceEvent, args);
```

Comment:
this looks to duplicate the code in the other service. if it's possible we should have it in a single place